inet_address_vectors.hh

```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include "gms/inet_address.hh"
+#include <vector>
+
+using inet_address_vector_replica_set = std::vector<gms::inet_address>;
+
+using inet_address_vector_topology_change = std::vector<gms::inet_address>;
```

Comment:
I fail to see the need for two types. What is special about the second one?

Reply:
Also why put inet_address_vector in there if you want to hide exactly those details behind a typedef? Later we may replace inet_address with host id (uuid) weill we do another rename?

Reply:
The mystery will be unveiled in the next patch - these types are identical now, but will differ in the space reserved for a small vector.

Reply:
So the reason for all the hoopla is to save 16 bytes on stack? And we want to allow arbitrary config changes anyway, so topology change will have more then one element.

Reply:
It's more than 16 bytes, since inet_address supports ipv6.

Reply:
It allows any size vector, this is just how much we optimize for.